vm/systemSmartContracts/delegation.go

```diff
@@ -284,18 +303,21 @@ func (d *delegation) init(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		TotalUnStaked:          big.NewInt(0),
 	}
 	globalFund.ActiveFunds[0] = fundKey
+	//TODO: If saveGlobalFundData will fail, then the last saves done will remain saved. Is this ok?
 	err = d.saveGlobalFundData(globalFund)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
+	//TODO: If saveDelegatorData will fail, then the last saves done will remain saved. Is this ok?
 	err = d.saveDelegatorData(ownerAddress, delegator)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
+	//TODO: If executeOnAuctionSC will fail, then the last saves done will remain saved. Is this ok?
```

Comment:
nothing will remain. Any error invalidates all previous changes. save is done in a cache - cache is invalidated if returnCode != vmcommon.Ok This is for all these comments.

Reply:
ok